package/MDAnalysis/core/universe.py

```diff
@@ -494,8 +495,22 @@ def transfer_to_memory(self, start=None, stop=None, step=None,
                 coordinates = []  # TODO: use pre-allocated array
                 for i, ts in enumerate(self.trajectory[start:stop:step]):
                     coordinates.append(np.copy(ts.positions))
+                    dimensions.append(np.copy(ts.dimensions))
                     pm.echo(i, frame=ts.frame)
                 coordinates = np.array(coordinates)
+            # If we extrated the coordinates using Timeseries, we still need
+            # to go through the trajectory to extract the dimensions.
+            else:
+                n_frames = len(range(
+                    *self.trajectory.check_slice_indices(start, stop, step)
+                ))
+                pm_format = ('{step}/{numsteps} frame dimensions '
+                             'copied to memory (frame {frame})')
+                pm = ProgressMeter(n_frames, interval=1,
+                                   verbose=verbose, format=pm_format)
+                for i, ts in enumerate(self.trajectory[start:stop:step]):
```

Comment:
This comes from an issue I have with Readers that implement `timeseries` (*i.e.* the DCD reader): the `timeseries` method only returns the coordinates. So whatever the method I use for to get the coordinates, I still need to go through the trajectory to get the box.

I see 3 ways of handling this:
1. a `timeseries` like method for the box to go with the one for the positions
2. having the box reading loginc twice: one if the `timeseries` is used that will iterate over the trajectory only to get the box, and one in the existing loop for readers that have no `timeseries`
3. what I did: having the logic only once, but iterating twice in some cases

Solution 1 is the best but implies to add a chunk to the reader, 2 the most efficient for little new development, 3 is the simplest and avoid logic duplication.

I am not sure what is the best approach here.